localstack/services/awslambda/lambda_api.py

```diff
@@ -1151,22 +1153,49 @@ def format_func_details(func_details, version=None, always_add_version=False):
 
 
 def forward_to_external_url(func_details, event, context, invocation_type):
+    func_forward_url = (
+        func_details.envvars.get("LOCALSTACK_LAMBDA_FORWARD_URL") or config.LAMBDA_FORWARD_URL
+    )
     """If LAMBDA_FORWARD_URL is configured, forward the invocation of this Lambda to the configured URL."""
-    if not config.LAMBDA_FORWARD_URL:
+    if not func_forward_url:
         return
+
     func_name = func_details.name()
     url = "%s%s/functions/%s/invocations" % (
-        config.LAMBDA_FORWARD_URL,
+        func_forward_url,
         PATH_ROOT,
         func_name,
     )
+
+    copied_env_vars = func_details.envvars.copy()
+    copied_env_vars["LOCALSTACK_HOSTNAME"] = config.HOSTNAME_EXTERNAL
+    copied_env_vars["LOCALSTACK_EDGE_PORT"] = str(config.EDGE_PORT)
+
     headers = aws_stack.mock_aws_request_headers("lambda")
+    headers["X-Amz-Region"] = func_details.region()
+    headers["X-Amz-Request-Id"] = context.aws_request_id
+    headers["X-Amz-Handler"] = func_details.handler
+    headers["X-Amz-Function-ARN"] = context.invoked_function_arn
+    headers["X-Amz-Function-Name"] = context.function_name
+    headers["X-Amz-Function-Version"] = context.function_version
+    headers["X-Amz-Role"] = func_details.role
+    headers["X-Amz-Runtime"] = func_details.runtime
+    headers["X-Amz-Timeout"] = str(func_details.timeout)
+    headers["X-Amz-Memory-Size"] = str(context.memory_limit_in_mb)
+    headers["X-Amz-Log-Group-Name"] = context.log_group_name
+    headers["X-Amz-Log-Stream-Name"] = context.log_stream_name
+    headers["X-Amz-Env-Vars"] = json.dumps(copied_env_vars)
+    headers["X-Amz-Last-Modified"] = str(int(func_details.last_modified.timestamp() * 1000))
     headers["X-Amz-Invocation-Type"] = invocation_type
     headers["X-Amz-Log-Type"] = "Tail"
-    client_context = context.get("client_context")
-    if client_context:
-        headers["X-Amz-Client-Context"] = client_context
-    data = json.dumps(event) if isinstance(event, dict) else str(event)
+    if context.client_context:
+        headers["X-Amz-Client-Context"] = context.client_context
+
+    def event_serializer(o):
+        if isinstance(o, datetime):
+            return o.isoformat()
+
+    data = json.dumps(event, default=event_serializer) if isinstance(event, dict) else str(event)
```

Comment:
nit: we could use `json_safe(..)` from `common.py` here:
```
data = json.dumps(json_safe(event))
```
(It uses `"%Y-%m-%dT%H:%M:%SZ"` as the date format, if that works for us here).

Reply:
@whummer 

Sure Done :+1: https://github.com/localstack/localstack/pull/4521/commits/c1d07bf2e8f92c47d0ba02801881db8a40ff4d1f

Just tried and tes it works. I was not aware of such common functionality :+1: